sceptre/resolvers/stack_output.py

```diff
@@ -169,11 +179,31 @@ def resolve(self):
         arguments = shlex.split(self.argument)
 
         stack_argument = arguments[0]
+
         if len(arguments) > 1:
-            extra_args = arguments[1].split("::", 2)
-            profile, region, sceptre_role = extra_args + (3 - len(extra_args)) * [None]
+            try:
+                extra_args = arguments[1].split("::", 2)
+
+                if len(extra_args) > 0:
+                    profile = extra_args[0]
+                if len(extra_args) > 1:
+                    region = extra_args[1]
+                if len(extra_args) > 2:
+                    sceptre_role = extra_args[2]
+
+            except ValueError as err:
+                message = (
+                    "!stack_output_external second arg should be "
+                    "in the format 'PROFILE[::REGION[::SCEPTRE_ROLE]]'"
+                )
+                raise SceptreException(message) from err
+
+        try:
+            dependency_stack_name, output_key = stack_argument.split("::")
+        except ValueError as err:
+            message = "!stack_output_external arg should match STACK_NAME::OUTPUT_KEY"
+            raise SceptreException(message) from err
```

Comment:
You know, this whole bit is probably a bit overly verbose. Consider this alternative:

```python
        arguments = iter(self.argument.split('::'))
        try:
            dependency_stack_name = next(arguments)
            output_key = next(arguments)
        except StopIteration:
            message = "!stack_output_external arg should match STACK_NAME::OUTPUT_KEY"
            raise SceptreException(message) from err
    
        profile = next(arguments, None)
        region = next(arguments, None)
        sceptre_role = next(arguments, None)
```

Reply:
Agree. This looks good after the refactor! Have a look now.